### DISCUSSION TITLE
Support for Diffractor AD?

### DISCUSSION THREAD

**Discussion by DominiqueMakowski** [TuringLang/Turing.jl#2100] [General]
As I believe faster sampling (especially for e.g., hierarchical models) will lead to an explosion of Turing and Julia adoption, I was wondering if there were any plans to support [Diffractor](https://github.com/JuliaDiff/Diffractor.jl) as an AD backend, and whether it might lead to speed gains.

I am aware that there have been some attempts at supporting Enzyme, but I'm not sure what is the estimated timeline for that.

**Comment by devmotion** [comment]
I don't have a feeling for how challenging it would be, one advantage could be that in contrast to Enzyme Diffractor uses ChainRules (another thing that slows down Enzyme adoption is that it is quite challenging to debug and deals with many LLVM internals, so not many people are knowledgable enough to fix issues). The first step would be to add support for Diffractor to https://github.com/tpapp/LogDensityProblemsAD.jl.
